components/filter/filter.js

```diff
@@ -412,10 +413,12 @@ class Filter extends LocalizeCoreElement(RtlMixin(LitElement)) {
 					this._totalAppliedCount--;
 				}
 			} else if (prop === 'values') {
-				if (dimension.searchType !== 'manual' || !dimension.searchValue) shouldRecount = true;
```

Comment:
When values are added/removed we can now always recount, and we'll need to re-search if a search was applied.